fastlane_bot/events/utils.py

```diff
@@ -560,6 +560,31 @@ def handle_flashloan_tokens(
     return flashloan_tokens
 
 
+def get_rpc_url(blockchain: str, tenderly_fork_id: str = None, rpc_url: str = None):
+    if tenderly_fork_id and not rpc_url:
+        rpc_url = f"https://rpc.tenderly.co/fork/{tenderly_fork_id}"
+    elif rpc_url == "" and blockchain == "ethereum":
+        WEB3_ALCHEMY_PROJECT_ID = os.environ.get("WEB3_ALCHEMY_PROJECT_ID")
+        rpc_url = f"https://eth-mainnet.alchemyapi.io/v2/{WEB3_ALCHEMY_PROJECT_ID}"
+    elif rpc_url == "" and blockchain == "polygon":
+        WEB3_ALCHEMY_POLYGON = os.environ.get("WEB3_ALCHEMY_POLYGON")
+        rpc_url = f"https://polygon-mainnet.g.alchemy.com/v2/{WEB3_ALCHEMY_POLYGON}"
+    elif rpc_url == "" and blockchain == "polygon_zkevm":
+        WEB3_ALCHEMY_POLYGON_ZKEVM = os.environ.get("WEB3_ALCHEMY_POLYGON_ZKEVM")
+        rpc_url = f"https://polygonzkevm-mainnet.g.alchemy.com/v2/{WEB3_ALCHEMY_POLYGON_ZKEVM}"
+    elif rpc_url == "" and blockchain == "optimism":
+        WEB3_ALCHEMY_OPTIMISM = os.environ.get("WEB3_ALCHEMY_OPTIMISM")
+        rpc_url = f"https://opt-mainnet.g.alchemy.com/v2/{WEB3_ALCHEMY_OPTIMISM}"
+    elif rpc_url == "" and blockchain == "coinbase_base":
+        WEB3_ALCHEMY_BASE = os.environ.get("WEB3_ALCHEMY_BASE")
+        rpc_url = f"https://base-mainnet.g.alchemy.com/v2/{WEB3_ALCHEMY_BASE}"
+    elif rpc_url == "" and blockchain == "arbitrum_one":
+        WEB3_ALCHEMY_ARBITRUM = os.environ.get("WEB3_ALCHEMY_ARBITRUM")
+        rpc_url = f"https://arb-mainnet.g.alchemy.com/v2/{WEB3_ALCHEMY_ARBITRUM}"
+
+    return rpc_url
```

Comment:
This function seems wrong in several different aspects:
1. Why does it need to take `rpc_url: str = None` as input to begin with?
2. Why does it check `not rpc_url` in some cases and `rpc_url == ""` in other cases?
3. Why does it check `rpc_url == ""` on every case? Surely there can be a simpler way
4. No final `else` case, is that why this function takes `rpc_url` as input to begin with?

It feels that there has to be a much less cumbersome way of writing this function, with the primary advantage being that it becomes less prone to unexpected behavior.

First off - no need to pass `rpc_url` just in order to return it as output if it contains a real value (i.e., neither `None` nor `""`).
Wherever this function is called, instead of passing that real `rpc_url` value, the caller can just use it instead of calling the function.
Subsequently everything becomes much cleaner in this function.
You can simply replace every `elif` statement with an `if` statement, and return the result directly from within each such statement.
And at the end of the function, in order to avoid unexpected behavior, you should raise an exception.